### PR TITLE
Add an API to decide what gets generated more granularly.

### PR DESCRIPTION
**Pull request by emilio**
This unifies options and is more flexible.

I'd prefer to do the `bitflags!` thing as a followup, because airport WiFi (it could be an E-Easy).

r? @fitzgen

### CODE REVIEW

File: src/options.rs [hunk 1]
```diff
@@ -149,11 +155,14 @@ pub fn builder_from_flags<I>(args: I)
...
-        return Err(Error::new(ErrorKind::Other, "Header not found"));
-    }
+    let header = match matches.value_of("header") {
+        Some(h) => h,
+        None => {
+            return Err(Error::new(ErrorKind::Other, "Header not found"));
+        }
+    };
+
+    builder = builder.header(header);
```
Comment (upsuper) [inline src/options.rs]: It seems to me this rewrite isn't necessary for this change, and actually the original way is more compact, and `header` has a clearer life time. I think it'd be better to just keep the original code here.

File: src/options.rs [hunk 2]
```diff
@@ -185,6 +194,23 @@ pub fn builder_from_flags<I>(args: I)
         }
     }
 
+    if let Some(what_to_generate) = matches.value_of("generate") {
```
Comment (upsuper) [inline src/options.rs]: It is probably better to state clearly what would happen if both `--generate` and one of `--ignore-*` options are specified. I guess it's probably easier to just reject that kind of combinations.

File: src/options.rs (hunk 1 above)
Comment (emilio) [inline src/options.rs]: Fair enough.

File: src/options.rs (hunk 2 above)
Comment (emilio) [inline src/options.rs]: Yeah, I agree with that, in general I want to clean all the options up once we start publishing this crate, so I decided to keep them for now... I thing it's not too unexpected to override with the `--ignore` flags.

**Comment by emilio** [comment]
Thanks for the review Xidorn :)

@bors-servo r=upsuper

**Pull request merged**